front/src/BottomBar/index.js

```diff
@@ -7,86 +7,149 @@ import faThLarge from "@fortawesome/fontawesome-free-solid/faThLarge";
 import React from "react";
 import PropTypes from "prop-types";
 import pageTypes from "./pageTypes";
+import connectionTypes from "./connectionTypes";
 
 /**
  * All of the props depend on the implementation in App.js. App.js should include WebSockets, with an onSend command,
  * which sends a command,a state variable which has the values "settings" and "refbox",
  * and the implementation of the functions passed to BottomBar.
  * Author: T.T.P. Franken
  * Author: B. Afonins
+ * Author: G.M. van der Sanden
  *
  * Props:
  * - onSend: function to send a command
  * - changeActivePage: function to change the active page
  * - activePage: a string indicating the current active page
+ * - isConnected: a boolean indicating whether the client is connected to the TRC
  */
 
 const BottomBar = props => {
-  const { changeActivePage, onSend, activePage } = props;
+  const { changeActivePage, onSend, activePage, isConnected } = props;
   const isSettingsPage = activePage === pageTypes.SETTINGS;
+  const connecting = isConnected === connectionTypes.CONNECTING;
+  const connected = isConnected === connectionTypes.CONNECTED;
   return (
-    <Bar className={props.className}>
-      <Button
-        id="bottom-bar__start-button"
-        onClick={() => onSend("start")}
-        enabled
+    <Bar>
+      <ConnectionBar
+        id="connectBar"
+        connecting={connecting}
+        connected={connected}
       >
-        <FontAwesomeIcon icon={faPlay} />
-      </Button>
-      <Button
-        id="bottom-bar__change-page-button"
-        onClick={() =>
-          changeActivePage(
-            isSettingsPage ? pageTypes.REFBOX : pageTypes.SETTINGS
-          )
-        }
-        enabled
-      >
-        {isSettingsPage ? (
-          <FontAwesomeIcon icon={faThLarge} />
-        ) : (
-          <FontAwesomeIcon icon={faCog} />
-        )}
-      </Button>
-      <Button
-        id="bottom-bar__stop-button"
-        onClick={() => onSend("stop")}
-        enabled
-      >
-        <FontAwesomeIcon icon={faStop} color="red" />
-      </Button>
+        <ConnectionStatus>
+          {connecting
+            ? connectionTypes.CONNECTING
+            : connected
+              ? connectionTypes.CONNECTED
+              : connectionTypes.DISCONNECTED}
```

Comment:
You can just pass isConnected. Also, eww, don't nest ternaries.

Reply:
...!